### PR TITLE
Child plugin messages should be reported hierarchically through parent plugin.

### PR DESCRIPTION
**Pull request by OrionsGuardian**
### Prerequisites

- [ ] Reviewed the [checklist](CHECKLIST.md)

- [ ] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

When viewing the plugin messages through the Plugin Reporter:
Child plugin messages will now be shown as part of a hierarchy of messages originating from a parent plugin.

### Alternate Designs

N/A

### Why Should This Be In Core?

For improved readability of plugin messages in the Plugin Reporter.

### Benefits

Seeing the hierarchy of messages in the Plugin Reporter provides an addition means of tracking the program flow.

### Possible Drawbacks

N/A

### Verification Process

Use the Analytic View to run some metrics on a graph.
Look at the plugin messages in the Plugin Reporter to confirm child messages are contained within parent messages.

Can also test in the Data Access View -> Test Parameters (Batched)
Confirm the child messages shown in the Plugin Reporter are contained within parent messages.

### Applicable Issues

#323

### CODE REVIEW

**Review by Delphinus8821 — APPROVED** [review]

File: CoreAnalyticView/src/au/gov/asd/tac/constellation/views/analyticview/AnalyticViewPane.java [hunk 1]
```diff
@@ -108,32 +122,62 @@ public AnalyticViewPane(final AnalyticController analyticController) {
...
-                        running = false;
-
-                        Platform.runLater(() -> {
-                            runButton.setText(RUN_START_TEXT);
-                            runButton.setStyle(RUN_START_STYLE);
-                        });
+                final Graph activeGraph = GraphManager.getDefault().getActiveGraph();
+                final SimplePlugin virtualAnalytics = new SimplePlugin("Analytic View - Query Runner"){
+                    @Override
+                    protected void execute(PluginGraphs graphs, PluginInteraction interaction, PluginParameters parameters) throws InterruptedException, PluginException {
```
Comment (antares1470) [inline CoreAnalyticView/src/au/gov/asd/tac/constellation/views/analyticview/AnalyticViewPane.java:128]: can you make the parameters final?

File: CoreAnalyticView/src/au/gov/asd/tac/constellation/views/analyticview/AnalyticViewPane.java [hunk 2]
```diff
@@ -108,32 +122,62 @@ public AnalyticViewPane(final AnalyticController analyticController) {
...
+                        questionThread.join();
+
+                        try {
+                            SwingUtilities.invokeAndWait(new Runnable(){
+                                @Override
+                                public void run() {
+                                    // waits for other queued tasks to complete, then does nothing and exits
+                                }
+                            });
+                        } catch (InvocationTargetException ex) {
```
Comment (antares1470) [inline CoreAnalyticView/src/au/gov/asd/tac/constellation/views/analyticview/AnalyticViewPane.java:169]: exception final

File: CoreAnalyticView/src/au/gov/asd/tac/constellation/views/analyticview/AnalyticViewPane.java [hunk 3]
```diff
@@ -108,32 +122,62 @@ public AnalyticViewPane(final AnalyticController analyticController) {
...
                     }
-                }, "Analytic View: Answer Question");
-                questionThread = answerQuestionThread;
-                answerQuestionThread.start();
+                };
+
+                try {
+                    PluginExecution.withPlugin(virtualAnalytics).interactively(true).executeNow(activeGraph);
+                } catch (InterruptedException | PluginException ex) {
+                    LOGGER.log(Level.SEVERE, ex.getLocalizedMessage());
```
Comment (antares1470) [inline CoreAnalyticView/src/au/gov/asd/tac/constellation/views/analyticview/AnalyticViewPane.java:179]: Can you separate the `InterruptedException` and add a `Thread.currentThread().interrupt()`?

**Review by antares1470 — CHANGES_REQUESTED** [review]
Looks good testing-wise

**Review by antares1470 — APPROVED** [review]

**Pull request merged**